db/document.go

```diff
@@ -132,7 +132,9 @@ func (doc *document) MarshalBody() ([]byte, error) {
 func unmarshalDocument(docid string, data []byte) (*document, error) {
 	doc := newDocument(docid)
 	if len(data) > 0 {
-		if err := json.Unmarshal(data, doc); err != nil {
+		decoder := json.NewDecoder(bytes.NewReader(data))
+		decoder.UseNumber()
+		if err := decoder.Decode(&doc); err != nil {
```

Comment:
`doc` is already a pointer, this is what is causing `TestImportNullDoc` to fail. produces incorrect values when data = []byte("null")

```go
if err := decoder.Decode(doc); err != nil { 
```

Reply:
Fixed